ci-operator/step-registry/upi/install/aws/cluster/upi-install-aws-cluster-commands.sh

```diff
@@ -0,0 +1,349 @@
+
+function gather_bootstrap_and_fail() {
+  if test -n "${GATHER_BOOTSTRAP_ARGS}"; then
+    openshift-install --dir=${ARTIFACT_DIR}/installer gather bootstrap --key "${SSH_PRIV_KEY_PATH}" ${GATHER_BOOTSTRAP_ARGS}
+  fi
+
+  return 1
+}
+# ensure LEASED_RESOURCE is set
+if [[ -z "${LEASED_RESOURCE}" ]]; then
+  echo "Failed to acquire lease"
+  exit 1
+fi
+
+cp "$(command -v openshift-install)" /tmp
+mkdir ${ARTIFACT_DIR}/installer
+
+echo "Installing from initial release ${RELEASE_IMAGE_LATEST}"
+SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+PULL_SECRET_PATH=${CLUSTER_PROFILE_DIR}/pull-secret
+OPENSHIFT_INSTALL_INVOKER=openshift-internal-ci/${JOB_NAME}/${BUILD_ID}
+AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred
+EXPIRATION_DATE=$(date -d '4 hours' --iso=minutes --utc)
+PULL_SECRET=${CLUSTER_PROFILE_DIR}/pull-secret
+OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="${RELEASE_IMAGE_LATEST}"
+
+export OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
+export PULL_SECRET_PATH
+export OPENSHIFT_INSTALL_INVOKER
+export AWS_SHARED_CREDENTIALS_FILE
+export EXPIRATION_DATE
+export PULL_SECRET
+
+mkdir -p ~/.ssh
+cp "${SSH_PRIV_KEY_PATH}" ~/.ssh/
+cp ${SHARED_DIR}/install-config.yaml ${ARTIFACT_DIR}/installer/install-config.yaml
+export PATH=${HOME}/.local/bin:${PATH}
+
+pushd ${ARTIFACT_DIR}/installer
+
+base_domain=$(python3 -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["baseDomain"])')
+AWS_REGION=$(python3 -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["platform"]["aws"]["region"])')
+CLUSTER_NAME=$(python3 -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["metadata"]["name"])')
+echo ${AWS_REGION} > ${SHARED_DIR}/AWS_REGION
+echo ${CLUSTER_NAME} > ${SHARED_DIR}/CLUSTER_NAME
+MACHINE_CIDR=10.0.0.0/16
+
+date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_START_TIME"
+openshift-install --dir=${ARTIFACT_DIR}/installer create manifests
+sed -i '/^  channel:/d' ${ARTIFACT_DIR}/installer/manifests/cvo-overrides.yaml
+rm -f ${ARTIFACT_DIR}/installer/openshift/99_openshift-cluster-api_master-machines-*.yaml
+rm -f ${ARTIFACT_DIR}/installer/openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+sed -i "s;mastersSchedulable: true;mastersSchedulable: false;g" ${ARTIFACT_DIR}/installer/manifests/cluster-scheduler-02-config.yml
+
+echo "Creating ignition configs"
+openshift-install --dir=${ARTIFACT_DIR}/installer create ignition-configs &
+wait "$!"
+
+cp ${ARTIFACT_DIR}/installer/bootstrap.ign ${SHARED_DIR}
+BOOTSTRAP_URI="https://${JOB_NAME_SAFE}-bootstrap-exporter-${NAMESPACE}.svc.ci.openshift.org/bootstrap.ign"
+export BOOTSTRAP_URI
+
+# begin bootstrapping
+if openshift-install coreos print-stream-json 2>/tmp/err.txt >coreos.json; then
+  RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.architectures.x86_64.images.aws.regions[$region].image' coreos.json)"
+else
+  RHCOS_AMI="$(jq -r .amis[\"${AWS_REGION}\"].hvm /var/lib/openshift-install/rhcos.json)"
```

Comment:
Does this have the same quoting issue as line 74?

Reply:
Good point. It doesn't but it looks inconsistent with the previous line so I'll change it